graph.py

```diff
@@ -0,0 +1,47 @@
+import random
+
+
+class Graph:
+    def __init__(self):
+        """
+        Graph variable is a dictionary of lists - graph is implemented as adjacency lists
+        """
+        self.graph = dict()
+
+    def _topological_sort(self, root, visited, result):
+        visited.add(root)
+        print(root)
+        for neighbour in self.graph[root]:
+            if neighbour not in visited:
+                result, visited = self._topological_sort(neighbour, visited, result)
```

Comment:
It looks like `result` and `visited` are modified in place so no need to return them explicitly?